File: 2025/src/day01/task1.rs
use std::{
    fs::File,
    io::{self, BufRead},
    time::Instant,
};

const PATH: &str = "inputs/day01/input.txt";

fn task1() {
    let mut dial_pos = 50;
    let mut zero_count = 0;

    let file = File::open(PATH).unwrap();
    let lines = io::BufReader::new(file).lines();

    for line in lines.map_while(Result::ok) {
        let direction: &str = &line[0..1];
        let amount: i32 = line[1..].parse().unwrap();
        match direction {
            "L" => dial_pos = (dial_pos - amount) % 100,
            "R" => dial_pos = (dial_pos + amount) % 100,
            &_ => panic!("Found invalid direction"),
        }

        if dial_pos < 0 {
            dial_pos += 100;
        }

        if dial_pos == 0 {
            zero_count += 1;
        }
    }
    println!("{zero_count}");
}

fn main() {
    let timer = Instant::now();
    task1();
    println!("Done in {:?}", timer.elapsed());
}

File: 2016/src/day02/task1/Task1.java
package day02.task1;

import java.nio.file.Files;
import java.nio.file.Path;
import java.time.Duration;
import java.time.Instant;

public class Task1 {
   Path path = Path.of("inputs", "day02", "input.txt");
   int row = 1, col = 1;
   StringBuilder result = new StringBuilder();

   void task() {
      try (var lines = Files.lines(this.path)) {
         lines.forEach((line) -> {
            for (var ch : line.toCharArray()) {
               switch (ch) {
                  case 'U' -> this.row = Math.clamp(this.row - 1, 0, 2);
                  case 'D' -> this.row = Math.clamp(this.row + 1, 0, 2);
                  case 'L' -> this.col = Math.clamp(this.col - 1, 0, 2);
                  case 'R' -> this.col = Math.clamp(this.col + 1, 0, 2);
               }
            }
            this.result.append(3 * this.row + this.col + 1);
         });

         System.out.println(this.result.toString());
      } catch (Exception e) {
         e.printStackTrace();
         System.exit(1);
      }
   }

   public static void main(String[] args) {
      var start = Instant.now();

      new Task1().task();

      System.out.println("Done in " + Duration.between(start, Instant.now()).toString().substring(2).toLowerCase());
   }
}
